ProcessLib/ThermoMechanicalPhaseField/CreateThermoMechanicalPhaseFieldProcess.cpp

```diff
@@ -0,0 +1,313 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateThermoMechanicalPhaseFieldProcess.h"
+
+#include <cassert>
+
+#include "MaterialLib/SolidModels/CreateLinearElasticIsotropic.h"
+#include "MaterialLib/SolidModels/LinearElasticIsotropicPhaseField.h"
+#include "ProcessLib/Output/CreateSecondaryVariables.h"
+
+#include "ThermoMechanicalPhaseFieldProcess.h"
+#include "ThermoMechanicalPhaseFieldProcessData.h"
+
+namespace ProcessLib
+{
+namespace ThermoMechanicalPhaseField
+{
+template <int DisplacementDim>
+std::unique_ptr<Process> createThermoMechanicalPhaseFieldProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "THERMO_MECHANICAL_PHASE_FIELD");
+    DBUG("Create ThermoMechanicalPhaseFieldProcess.");
+
+    auto const staggered_scheme =
+        //! \ogs_file_param{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__coupling_scheme}
+        config.getConfigParameterOptional<std::string>("coupling_scheme");
+    const bool use_monolithic_scheme =
+        !(staggered_scheme && (*staggered_scheme == "staggered"));
+    if (!(staggered_scheme && (*staggered_scheme == "staggered")))
+    {
+        OGS_FATAL(
+            "Monolithic scheme for thermo-mechanical coupled "
+            "phase field process is currently not available.");
+    }
+
+    // Process variable.
+
+    //! \ogs_file_param{prj__processes__process__THERMO_MECHANICAL_PHASE_FIELD__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+    ProcessVariable* variable_T;
+    ProcessVariable* variable_ph;
+    ProcessVariable* variable_u;
+    std::vector<std::vector<std::reference_wrapper<ProcessVariable>>>
+        process_variables;
+    int mechanics_related_process_id = 0;
+    int phase_field_process_id = 0;
+    int heat_conduction_process_id = 0;
+    if (use_monolithic_scheme)  // monolithic scheme.
```

Comment:
Since you did not implement the monolithic scheme, I'd replace these code parts with `OGS_FATAL` calls with explanatory messages.

Reply:
This is still not addressed.

Reply:
An OGS_FATAL is already in line 43 for  the monolithic scheme. If the monolithic scheme is not interesting for solving of coupled TMPh problems , all monolithic scheme related stuff in this PR can be removed.

Reply:
I agree with Wenqing, judging by my experience on testing specific cases for model verification and applications, the monolithic scheme is not an efficient method for solving complex cases.